odc/stats/io.py

```diff
@@ -347,13 +347,19 @@ def get_eo3_stac_meta(
         )  # stac_meta is Python str, but content is 'Dict format'
 
     def dump_with_pystac(
-        self, task: Task, ds: Dataset, aux: Dataset | None = None
+        self,
+        task: Task,
+        ds: xr.Dataset,
+        aux: xr.Dataset | None = None,
+        proc: StatsPluginInterface = None,
```

Comment:
I somehow managed to comment on one of the intermediate commits, second attempt:

`StatsPluginInterface` is a class, so that can't be None.